modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/MoveModal.js

```diff
@@ -49,59 +38,18 @@ export default function MoveModal({
 	const [searchActive, setSearchActive] = useState(false);
 
 	const handleItemMove = (item, parentItem, index) => {
-		if (
-			item.type === ITEM_TYPES.folder &&
-			parentItem.type === ITEM_TYPES.article
-		) {
-			openToast({
-				message: Liferay.Language.get(
-					'folders-cannot-be-moved-into-articles'
-				),
-				type: 'danger',
-			});
-
-			return false;
-		}
-
-		fetch(moveKBObjectURL, {
-			body: objectToFormData({
-				[`${portletNamespace}dragAndDrop`]: true,
-				[`${portletNamespace}position`]: index?.next ?? -1,
-				[`${portletNamespace}resourceClassNameId`]: item.classNameId,
-				[`${portletNamespace}resourcePrimKey`]: item.id,
-				[`${portletNamespace}parentResourceClassNameId`]: parentItem.classNameId,
-				[`${portletNamespace}parentResourcePrimKey`]: parentItem.id,
-			}),
-			method: 'POST',
-		})
-			.then((response) => {
-				if (!response.ok) {
-					throw new Error();
-				}
-
-				return response.json();
-			})
-			.then((response) => {
-				if (!response.success) {
-					throw new Error(response.errorMessage);
-				}
-
-				showSuccessMessage(portletNamespace);
-			})
-			.catch(
-				({
-					message = Liferay.Language.get(
-						'an-unexpected-error-occurred'
-					),
-				}) => {
-					openToast({
-						message,
-						type: 'danger',
-					});
-				}
-			);
-
-		return true;
+		getOpener().Liferay.fire('selectKBMoveFolder', {
```

Comment:
No need to send the item here. Only the `index` and the `selectedItem`. 
Also, use a constant for the eventName (declare outside the export fn a `const SELECT_EVENT_NAME='selectKBMoveFolder'`) and fire this `getOpener().Liferay.fire(SELECT_EVENT_NAME, ...`